api/core/v1beta1/machinehealthcheck_types.go

```diff
@@ -69,6 +69,14 @@ type MachineHealthCheckSpec struct {
 	// +kubebuilder:validation:MaxItems=100
 	UnhealthyConditions []UnhealthyCondition `json:"unhealthyConditions,omitempty"`
 
+	// unhealthyMachineConditions contains a list of the machine conditions that determine
```

Comment:
Currently not enough bandwidth for a full review, but a few quick comments already:
* we should only add this to v1beta2
* we have to add this in a few places. I would recommend to rebase on top of main and search for all occurences of "nodeUnhealthyConditions" and check in which places we also need "unhealthyMachineConditions" (it will be in a lot of these cases :))